pkg/dotnet/core_deps/parse.go

```diff
@@ -31,30 +50,60 @@ func (p *Parser) Parse(r dio.ReadSeekerAt) ([]types.Library, []types.Dependency,
 	}
 
 	var libraries []types.Library
-	for nameVer, lib := range depsFile.Libraries {
-		if !strings.EqualFold(lib.Type, "package") {
+	var deps []types.Dependency
+	for pkgNameVersion, target := range depsFile.Targets[depsFile.RuntimeTarget.Name] {
+		library, ok := depsFile.Libraries[pkgNameVersion]
```

Comment:
Can we discuss about this:

It looks like we can take package name and version from `pkgNameVersion`.
Do we need to check "Libraries"?
Is this necessary for backwards compatibility?

Reply:
I think I did this initially because I wanted to keep track of whether a Library was a project or not and then be able to set the CycloneDX `componentType` in response, but I don't think go-dep-parser keeps track of this? I don't see any field on the `Library` struct that would allow us to set something resembling `componentType`

Reply:
So, in this case, we don't need to check Libraries at all

Reply:
> I don't think go-dep-parser keeps track of this

We set `componentType` in Trivy. 

>  I wanted to keep track of whether a Library was a project or not 

We currently take only packages - https://github.com/aquasecurity/go-dep-parser/blob/1237b47625e1c1e543dcc899e112c3b1e3d8b7ff/pkg/dotnet/core_deps/parse.go#L35
For project we use `application` type with filepath as name:
```
➜  trivy -d fs -f cyclonedx pkg/dotnet/core_deps/testdata/ExampleApp1.deps.json | jq '.components[] | select(.type == "application")'
...
{
  "bom-ref": "908ce44b-3644-44ce-b282-8f0d44566e20",
  "type": "application",
  "name": "ExampleApp1.deps.json",
  "properties": [
    {
      "name": "aquasecurity:trivy:Class",
      "value": "lang-pkgs"
    },
    {
      "name": "aquasecurity:trivy:Type",
      "value": "dotnet-core"
    }
  ]
}
```